cassiopeia/tools/small_parsimony.py

```diff
@@ -239,3 +241,235 @@ def score_small_parsimony(
                 "infer_ancestral_states=True."
             )
     return parsimony
+
+
+def fitch_count(
+    cassiopeia_tree: CassiopeiaTree,
+    meta_item: str,
+    root: Optional[str] = None,
+    infer_ancestral_states: bool = True,
+    state_key: str = "S1",
+):
+    """Runs the FitchCount algorithm.
+
+    Performs the FitchCount algorithm for inferring the number of times that
+    two two states transition to one another across all equally-parsimonious
+    solutions returned by the Fitch-Hartigan algorithm. The original algorithm
+    was described in Quinn, Jones, et al, Science (2021). The output is an 
+    MxM count matrix, where the values indicate the number of times that
+    m1 transitioned to m2 along an edge in a Fitch-Hartigan solution.
+    To obtain probabilities P(m1 -> m2), divide each row by its row-sum.
+
+    This procedure will only work on categorical data and will otherwise raise
+    an error.
+
+    Args:
+        cassiopeia_tree: CassiopeiaTree object with a tree and cell meta data.
+        meta_item: A column in the CassiopeiaTree cell meta corresponding to a
+            categorical variable.
+        root: Node to treat as the root.
+        infer_ancestral_states: Whether or not to initialize the ancestral state
+            sets with Fitch-Hartigan.
+        state_key: If ancestral state sets have already been created, then this
+            argument specifies what the attribute name is in the CassiopeiaTree
+
+    Returns:
+        An MxM count matrix indicating the number of edges that contained a
+            transition between two states across all equally parsimonious
+            solutions returned by Fitch-Hartigan.
+    """
+    cassiopeia_tree = cassiopeia_tree.copy()
+
+    unique_states = cassiopeia_tree.cell_meta[meta_item].unique()
```

Comment:
Instead of inferring the states from the table, why not have the user provide them optionally, to cover the case that there is a state with zero instances? This might sound like a strange use case, but for rare states that appear on some trees and not on others I think it's nice for M to be the same for all trees.

Reply:
Not a bad idea, I'll add this as an option.